sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/TooManyParameters.cs

```diff
@@ -0,0 +1,118 @@
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class TooManyParameters : TooManyParametersBase<SyntaxKind, ParameterListSyntax>
+    {
+        protected override GeneratedCodeRecognizer GeneratedCodeRecognizer => VisualBasicGeneratedCodeRecognizer.Instance;
+        protected override SyntaxKind[] SyntaxKinds => new SyntaxKind[] { SyntaxKind.ParameterList };
+        protected override DiagnosticDescriptor Rule => rule;
+        protected override Dictionary<SyntaxKind, string> Mapping => mapping;
+        protected override SyntaxKind ParentType(ParameterListSyntax parameterList) => parameterList.Parent.Kind();
+        protected override int CountParameters(ParameterListSyntax parameterList) => parameterList.Parameters.Count;
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+
+        protected override bool CanBeChanged(SyntaxNode node, SemanticModel semanticModel)
+        {
+            var declaredSymbol = semanticModel.GetDeclaredSymbol(node);
+            var symbol = semanticModel.GetSymbolInfo(node).Symbol;
+
+            if (node.IsAnyKind(LambdaHeaders))
+            {
+                return true;
+            }
+
+            if (declaredSymbol == null && symbol == null)
+            {
+                // no information
+                return false;
+            }
+
+            if (symbol != null)
+            {
+                // Not a declaration, such as Action
+                return true;
+            }
+
+            if ((node as SubNewStatementSyntax)?.ParameterList?.Parameters.Count is int parameterCount &&
+                parameterCount > Maximum &&
+                node.Parent is ConstructorBlockSyntax constructorBlock &&
+                ContainsMyBaseNewInvocation(constructorBlock, Maximum))
+            {
+                return false;
+            }
+
+            if (declaredSymbol.IsExtern &&
+                declaredSymbol.IsStatic &&
+                declaredSymbol.GetAttributes(KnownType.System_Runtime_InteropServices_DllImportAttribute).Any())
+            {
+                // P/Invoke method is defined externally.
+                // Do not raise
+                return false;
+            }
+
+            return declaredSymbol.GetOverriddenMember() == null &&
+                   declaredSymbol.GetInterfaceMember() == null;
+        }
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager,
+                isEnabledByDefault: false);
+
+        private static bool ContainsMyBaseNewInvocation(ConstructorBlockSyntax constructorBlock, int maximum) =>
+                constructorBlock.Statements.Any(s => s is ExpressionStatementSyntax expression &&
+                    expression.Expression is InvocationExpressionSyntax invocation &&
```

Comment:
No need for this indentation for this and the other statements.

Reply:
all expressions are an argument for the `Any` call, I prefer to indent to make it clear